trikot-streams/streams/src/commonMain/kotlin/com/mirego/trikot/streams/cancellable/CancellableManager.kt

```diff
@@ -28,8 +30,22 @@ class CancellableManager : Cancellable {
         )
     }
 
+    fun cleanCancelledChildren() {
```

Comment:
Thinking out loud here. I wonder if that's not something we should do automatically in the `add()`

Reply:
Yeah.. I don't see why not, that would keep it leaner and cleaner at all times, making potential manual cleanups faster.

Reply:
After discussion we determined that it might be too heavy to execute automatically in the `add()` if the `queueList` is really big. We can have a lot of `Cancellable` that will never be cleanable in the list, so you'll have to iterate through them every time for nothing.

So ideally solution is to let the owner of the `CancellableManager` call `cleanCancelledChildren()` at the right moment.